torch/_inductor/codegen/triton.py

```diff
@@ -142,7 +142,8 @@ def to_dtype(x, dtype: torch.dtype):
 
     @staticmethod
     def constant(value, dtype):
-        return triton_constant(value)
+        type_ = torch._prims_common.dtype_to_type(dtype)
+        return triton_constant(type_(value))
```

Comment:
This still doesn't respect the size of `dtype`, e.g. 300 could pass through `torch.int8` without being truncated. However, that would be undefined behavior in C++ anyway so I'm not sure it matters. Though this may be worth thinking about for  `uint8`.

Reply:
I'd say that it'd be good to have a check as it may uncover issues. Worst comes worst, we could put it behind a debug flag.